### PR TITLE
Normalize molecule infrastructure

### PR DESCRIPTION
**Pull request by Deydra71**
Create basic structure for generating molecule tests.

### CODE REVIEW

**Comment by Deydra71** [comment]
Currently correcting the generation of prepare.yml files, as the tests are failing due to the change of the path of test_deps.

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

File: scripts/generate_molecule.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+import os
+import yaml
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+
+template_path = "molecule/common"
```
Comment (jpodivin) [inline scripts/generate_molecule.py:5]: Constants should be in upper case.

File: scripts/generate_molecule.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    'scenario': {'name': ''}
+                }
+
+        if file == 'collections.j2':
+            collections_list = ['community.general']
+            collections_formatted = "\n- ".join(collections_list)
+            render_config = {"collections": collections_formatted}
+
+        if file == 'prepare.j2':
+            if 'Prepare test_deps' in existing_prepare_content or 'role: ../../../../molecule/common/test_deps' in existing_prepare_content:
```
Comment (jpodivin) [inline scripts/generate_molecule.py:84]: The later string could be factored out as constant to improve readability.

**Review by jpodivin — CHANGES_REQUESTED** [review]
It's mostly great. Apart from the few things in the script, I think it would be a good idea to write some comments there as well.

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by Deydra71** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Comment by danpawlik** [comment]
recheck

**Review by jpodivin — APPROVED** [review]

**Pull request merged**